src/java.base/linux/native/libnio/fs/LinuxNativeDispatcher.c

```diff
@@ -199,6 +199,7 @@ Java_sun_nio_fs_LinuxNativeDispatcher_directCopy0
                     case EINVAL:
                     case ENOSYS:
                     case EXDEV:
+                    case EPERM:
```

Comment:
This is unrelated to [JDK-8316304](https://bugs.openjdk.org/browse/JDK-8316304). If you want to fix this as well, then please as a separate patch. The bug speaks about JDK-8316304 and JDK-8316304 only.